drivers/adc/ad400x/ad400x.c

```diff
@@ -136,33 +128,32 @@ int32_t ad400x_spi_single_conversion(struct ad400x_dev *dev,
  * @return 0 in case of success, negative error code otherwise.
  */
 int32_t ad400x_init(struct ad400x_dev **device,
-		    struct ad400x_init_param init_param)
+		    struct ad400x_init_param *init_param)
 {
 	struct ad400x_dev *dev;
 	int32_t ret;
 	uint8_t data = 0;
+	struct spi_engine_init_param *spi_eng_init_param = init_param->spi_init.extra;
 
 	dev = (struct ad400x_dev *)malloc(sizeof(*dev));
 	if (!dev)
 		return -1;
 
-	ret = spi_init(&dev->spi_desc, init_param.spi_init);
+	ret = spi_init(&dev->spi_desc, &init_param->spi_init);
 	if (ret < 0)
 		goto error;
 
-	dev->dev_id = init_param.dev_id;
-	dev->spi_desc->cs_delay = init_param.spi_init.cs_delay;
+	dev->reg_access_speed = init_param->reg_access_speed;
+	dev->dev_id = init_param->dev_id;
 
-	dev->spi_desc->max_data_width = init_param.num_bits;
-
-	spi_set_transfer_length(dev->spi_desc, 16);
+	spi_engine_set_transfer_width(dev->spi_desc, 16);
```

Comment:
this is not 16, it depends on the number of bits of the ADC, with AD4020 it is 20.